waspc/src/Wasp/Analyzer/Parser/Tokenizer.hs

```diff
@@ -0,0 +1,26 @@
+module Wasp.Analyzer.Parser.Tokenizer
+  ( isValidWaspIdentifier,
+  )
+where
+
+import Control.Monad.Except (runExcept)
+import Control.Monad.State (evalStateT)
+import Wasp.Analyzer.Parser.Lexer (alexScanTokens)
+import Wasp.Analyzer.Parser.Monad (Parser, makeInitialState)
+import Wasp.Analyzer.Parser.Token (Token (Token), TokenType (TIdentifier))
+
+isValidWaspIdentifier :: String -> Bool
+isValidWaspIdentifier str = case tokenize str of
+  Just [Token (TIdentifier name) _ _] -> length name == length str
```

Comment:
The `length` part ensures nothing was skipped, but it's a bit weird. I'm open to better ideas.

Reply:
Ha yeah that is a bit weird. Do you really need to check this? If we scanned the token, isn't that enough? But you want to check if we skipped anything.
You can do `let wasAnyCharSkipped = ... in wasAnyCharSkipped` to possibly document this part.

Reply:
> If we scanned the token, isn't that enough? But you want to check if we skipped anything.
Yes, otherwise it can create projects with names like "      something "

> You can do let wasAnyCharSkipped = ... in wasAnyCharSkipped to possibly document this part.
I like the idea of using variables for documentation purposes, but have a strong feeling that the linter will complain with something like "redundant let expression". We'll see tomorrow.